contour_node/gaussian[] functions